org.adempiere.base/src/org/adempiere/model/GenericPO.java

```diff
@@ -132,6 +146,604 @@ protected int get_AccessLevel() {
 		return Integer.parseInt(p_info.getAccessLevel());
 	}
 
+
+
+	/* METHODS TO USE GenericPO in Documents */
+
+	/**
+	 * Verify if a column exists
+	 * @param columnName
+	 * @param throwException - must throw an exception when the column doesn't exist
+	 * @return
+	 */
+	private boolean columnExists(String columnName, boolean throwException) {
+		int idx = get_ColumnIndex(columnName);
+		if (idx < 0 && throwException)
+			throw new AdempiereException("Column " + get_TableName() +"." + columnName + " not found");
+		return (idx >= 0);
+	}
+
+	/**
+	 * Verify if a column exists
+	 * @param columnName
+	 * @return boolean
+	 */
+	private boolean columnExists(String columnName) {
+		return columnExists(columnName, false);
+	}
+
+	/** Column name DocAction */
+	public static final String COLUMNNAME_DocAction = "DocAction";
+	/** Column name DocStatus */
+	public static final String COLUMNNAME_DocStatus = "DocStatus";
+	/** Column name Processing */
+	public static final String COLUMNNAME_Processing = "Processing";
+	/** Column name C_DocTypeTarget_ID */
+	public static final String COLUMNNAME_C_DocTypeTarget_ID = "C_DocTypeTarget_ID";
+	/** Column name C_DocType_ID */
+	public static final String COLUMNNAME_C_DocType_ID = "C_DocType_ID";
+	/** Column name DateAcct */
+	public static final String COLUMNNAME_DateAcct = "DateAcct";
+	/** Column name DateTrx */
+	public static final String COLUMNNAME_DateTrx = "DateTrx";
+	/** Column name IsApproved */
+	public static final String COLUMNNAME_IsApproved = "IsApproved";
+	/** Column name Processed */
+	public static final String COLUMNNAME_Processed = "Processed";
+	/** Column name DocumentNo */
+	public static final String COLUMNNAME_DocumentNo = "DocumentNo";
+	/** Column name Description */
+	public static final String COLUMNNAME_Description = "Description";
+	/** Column name SalesRep_ID */
+	public static final String COLUMNNAME_SalesRep_ID = "SalesRep_ID";
+	/** Column name AD_User_ID */
+	public static final String COLUMNNAME_AD_User_ID = "AD_User_ID";
+	/** Column name C_Currency_ID */
+	public static final String COLUMNNAME_C_Currency_ID = "C_Currency_ID";

```

Comment:
similar to the constants below, it seems more intuitive to have this column constants at the DocAction interface instead.

Reply:
done - I needed to add a prefix DOC_COLUMNNAME - otherwise it conflicts with the actual constants